format/spec.md

```diff
@@ -298,16 +298,101 @@ Iceberg tables must not use field ids greater than 2147483447 (`Integer.MAX_VALU
+| **`2147483644  _deleted`**       | `boolean`     | Whether the row has been deleted                                                                        |
+| **`2147483643  _spec_id`**       | `int`         | Spec ID used to track the file containing a row                                                         |
+| **`2147483642  _partition`**     | `struct`      | Partition to which a row belongs                                                                        |
+| **`2147483546  file_path`**      | `string`      | Path of a file, used in position-based delete files                                                     |
+| **`2147483545  pos`**            | `long`        | Ordinal position of a row, used in position-based delete files                                          |
+| **`2147483544  row`**            | `struct<...>` | Deleted row values, used in position-based delete files                                                 |
+| **`2147483543  _row_id`**        | `long`        | A unique long assigned when row-lineage is enabled see [Row Lineage](#row-lineage)                      |
+| **`2147483542  _last_update`**   | `long`        | The sequence number which last updated this row when row-lineage is enabled [Row Lineage](#row-lineage) |
+
+### Row Lineage
+
+In v3 and later, an Iceberg table can track row lineage fields for all newly created rows.  Row lineage is enabled by setting the field `row-lineage` to true in the table's metadata. When enabled, engines must maintain the `next-row-id` table field and the following row-level fields when writing data files:
+
+* `_row_id` a unique long for every row. Assigned via inheritance when a row is added and the existing value is explicitly written when the row is written to a new file.
+* `_last_update` the sequence number of the commit that last updated a row. The value is inherited when a row is added or modified and the existing value is explicitly written when the row is written to a different data file but not modified.
+
+These fields are assigned and updated by inheritance because the commit sequence number and starting row ID are not assigned until the snapshot is successfully committed. Inheritance is used to allow writing data and manifest files before values are known so that it is not necessary to rewrite data and manifest files when an optimistic commit is retried.
+
+When row lineage is enabled, new snapshots cannot include [Equality Deletes](#equality-delete-files). Row lineage is incompatible with equality deletes because lineage values must be maintained, but equality deletes are used to avoid reading existing data before writing changes.
+
+
+#### Row lineage assignment
+
+Row lineage fields are written when row lineage is enabled. When not enabled, row lineage fields (`_row_id` and `_last_update`) must not be written to data files. The rest of this section applies when row lineage is enabled.
```

Comment:
The same clean-up question here. Do we rewrite data files in case of disabling row lineage or we disallow disabling?

Reply:
I think disabling is allowed it just means we stop changing any of the metadata and lineage information may be possibly lost by clients which don't support row-lineage. I don't think we need to prevent this.